tracer/src/Datadog.Trace/Iast/SensitiveData/HeaderInjectionTokenizer.cs

```diff
@@ -0,0 +1,49 @@
+// <copyright file="HeaderInjectionTokenizer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Configuration;
+
+#nullable enable
+
+namespace Datadog.Trace.Iast.SensitiveData;
+
+/// <summary>
+/// We are already redacting sensitive values in the evidence (header values),
+/// but if the key of the returned header matches a regex, we should also redact the evidence value after ":"
+/// </summary>
+internal class HeaderInjectionTokenizer : ITokenizer
+{
+    internal const string KeysRegex = @"(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?|access_?|secret_?)key(?:_?id)?|token|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)";
```

Comment:
I'd put the string in the Regex constructor and skip the Java "pattern". The regex is not the regex, it's a string, and the real regex is called pattern. It's a bit confusing.

Reply:
Updated.